crates/subspace-networking/src/request_handlers/generic_request_handler.rs

```diff
@@ -52,14 +53,17 @@ pub struct GenericRequestHandler<Request: GenericRequest> {
 }
 
 impl<Request: GenericRequest> GenericRequestHandler<Request> {
-    pub fn create<RH, Fut>(request_handler: RH) -> Box<dyn RequestHandler>
+    pub fn create<RH, Fut>(protocol_prefix: String, request_handler: RH) -> Box<dyn RequestHandler>
```

Comment:
I'm not sure I follow your comment about the previous implementation, also we use the protocol name from the config for the `RequestResponseBehaviour`. Did you mean adding `protocol_name()` function to `GenericRequest` trait and using it during the protocol creation?

Reply:
> Did you mean adding protocol_name() function to GenericRequest trait and using it during the protocol creation?

I didn't check the code very carefully, but I think no. Config stays the same, but when you get to actually use it and create request/response protocol, you can add a prefix to it. At the very end instead of very beginning. You can then do the same for Kademlia and Gossipsub, all will be prefixed at the end, but everywhere else they can stay the way they are.

Reply:
I ran into unexpected issues with request-response protocol tests. It should be a standalone task: https://github.com/subspace/subspace/issues/1373